test/runtests.jl

```diff
@@ -289,6 +289,20 @@ end
 @test printpoly_to_string(Poly([1,2,3], "y")) == "1 + 2*y + 3*y^2"
 @test printpoly_to_string(Poly([1,2,3], "y"), descending_powers=true) == "3*y^2 + 2*y + 1"
 
+@test printpoly_to_string(Poly([1.0,2.0,3.0], "y")) == "1.0 + 2.0*y + 3.0*y^2"
+
+@test printpoly_to_string(Poly([1//2, 2//3], "y")) == "1//2 + 2//3*y"
+
+@test printpoly_to_string(Poly([Inf*im, Inf*im, NaN, NaN*im])) == "Inf*im + Inf*im*x + NaN*x^2 + NaN*im*x^3"
+@test printpoly_to_string(Poly([1 + 1im])) == "(1 + 1im)"
+@test printpoly_to_string(Poly([1im, 1im])) == "im + im*x"
+@test printpoly_to_string(Poly([1 + im, 2 + 4im])) == "(1 + 1im) + (2 + 4im)*x"
+@test printpoly_to_string(Poly([1.0 + im, 2 + 4im])) == "(1.0 + 1.0im) + (2.0 + 4.0im)*x"
+
+using DualNumbers
```

Comment:
This will require adding `DualNumbers` to test/REQUIRE? It might be best if you could make a simple type on the fly that shows your behaviour rather than pull in the dependency through CI, though it seems like a light one.

Reply:
Tried for a little bit, but seems like there are quite a few operations that would have to be defined for the that type. I guess there are also some benefits of testing for one of the actual number type that is of interest.